zap/src/test/java/org/zaproxy/zap/model/SessionStructureUnitTest.java

```diff
@@ -61,6 +62,14 @@ public void setUp() throws Exception {
         given(model.getVariantFactory()).willReturn(factory);
         Control.initSingletonForTesting(model);
         msg = new HttpMessage();
+        sitesNode = mock(StructuralSiteNode.class);
+        given(sitesNode.isRoot()).willReturn(true);
+        hostNode = mock(StructuralSiteNode.class);
+        given(hostNode.getParent()).willReturn(sitesNode);
+        given(hostNode.isLeaf()).willReturn(true);
+        URI uri = new URI("https://www.example.com", true);
+        given(hostNode.getURI()).willReturn(uri);
+        given(hostNode.getName()).willReturn("https://www.example.com");
```

Comment:
The new tests could be done in a nested class, since this is not required for all of them.